docs/developer-guide/firstanalysis.md

```diff
@@ -0,0 +1,144 @@
+# Your first analysis
+
+This is a tutorial exercise for writing analyses in Goblint.
+We will implement a very simple sign analysis.
+You need to install goblint for development.
+
+## First test
+
+Create a C file in the Goblint root directory for testing.
+Here we call it `example.c`. This is the first test:
+
+
+```
+#include<assert.h>
+
+int main() {
+
+  int x;
+  int unknown;
+
+  if (unknown) {
+    x = 5;
+  } else {
+    x = 7;
+  }
+
+  // The above code branches on an uninitialized variable.
+  // The value of x could be either 5 or 7.
+
+  assert(x > 0);  // Thus, this assertion should hold!
+
+  return 0;
+}
+```
+
+If you run goblint out of the box on this example it will not work: `./goblint --enable dbg.debug example.c` will claim that the assertion in unknown.
+Goblint could verify that this asertion does hold using interval analysis (`--enable ana.int.interval`), but here we will implement a simple sign analysis instead.
+
+We begin by registering and running a new analysis called "signs":
+
+1. Create a copy of the unit analysis (*src/analyses/unit.ml*) into the file *src/analyses/signs.ml*.
+2. Inside the file, change the name from "unit" to "signs".
+3. Try to run goblint with the new analysis enabled: `./goblint --sets "ana.activated[+]" signs --enable dbg.debug example.c`. The result will still be that nothing is verified.
+4. Now edit the analysis to pass the test. :)
+
+
+## Designing the domain
+
+The last step might need some clarifications. We first need to design the abstract domain. It may help if you have read some theoretical tutorial on abstract domains. Our first sign lattice will simply contain the elements `{-, 0, +}` with top and bottom added. For this, we define first the signs and then we lift it. Insert the following into your file just before the module `Spec` is defined.
+
+```
+module Signs =
+struct
+  type t = Neg | Zero | Pos [@@deriving eq, to_yojson]
+  let name () = "signs"
+  let show x = match x with
+  | Neg -> "-"
+  | Pos -> "+"
+  | Zero -> "0"
+
+  (* We need the following to generate output... *)
+  include Printable.StdPolyCompare
```

Comment:
Shouldn't recommend `StdPolyCompare`, but just `Std` and use `[@@deriving ord]` instead.

Reply:
This is a simple fix for this tutorial, but why haven't we deleted StdPolyCompare? I just copy-pasted this from somewhere. :)

Reply:
Yes, `StdPolyCompare` and `BlankPolyCompare` being gone is great! Less possibility for bugs!